src/AspNetCoreModuleV2/AspNetCore/applicationinfo.h

```diff
@@ -19,150 +17,66 @@ class APPLICATION_INFO
 {
 public:
 
-    APPLICATION_INFO(IHttpServer &pServer) :
+    APPLICATION_INFO(
+        IHttpServer &pServer,
+        IHttpApplication    &pApplication,
+        HandlerResolver     *pHandlerResolver
+    ) :
         m_pServer(pServer),
-        m_cRefs(1),
-        m_handlerResolver(nullptr)
+        m_handlerResolver(pHandlerResolver),
+        m_struConfigPath(pApplication.GetAppConfigPath()),
+        m_struInfoKey(pApplication.GetApplicationId())
     {
         InitializeSRWLock(&m_applicationLock);
     }
 
-    PCWSTR
+    std::wstring&
     QueryApplicationInfoKey()
     {
-        return m_struInfoKey.QueryStr();
+        return m_struInfoKey;
     }
 
-    STRU*
+    std::wstring&
     QueryConfigPath()
     {
-        return &m_struConfigPath;
+        return m_struConfigPath;
     }
 
     virtual
     ~APPLICATION_INFO();
 
-    static
-    void
-    StaticInitialize()
-    {
-    }
+    VOID
+    ShutDownApplication(bool fServerInitiated);
 
     HRESULT
-    Initialize(
-        IHttpApplication    &pApplication,
-        HandlerResolver     *pHandlerResolver
+    GetOrCreateApplication(
+        IHttpContext *pHttpContext,
+        std::unique_ptr<IAPPLICATION, IAPPLICATION_DELETER>& pApplication
     );
 
-    VOID
-    ReferenceApplicationInfo() const
-    {
-        InterlockedIncrement(&m_cRefs);
-    }
-
-    VOID
-    DereferenceApplicationInfo() const
+    bool ConfigurationPathApplies(const std::wstring& path)
     {
-        if (InterlockedDecrement(&m_cRefs) == 0)
+        // We need to check that the last character of the config path
+        // is either a null terminator or a slash.
+        // This checks the case where the config path was
+        // MACHINE/WEBROOT/site and your site path is MACHINE/WEBROOT/siteTest
+        auto const changed = m_struConfigPath._Starts_with(path);
```

Comment:
where does `_Starts_with` come from?

Reply:
microsoft implementation of std::wstring